### PR TITLE
Setup Parser to allow Custom Tags

### PR DESCRIPTION
**Pull request by Spadavecchia**

### CODE REVIEW

File: lib/liquid/combinators/general.ex [hunk 1]
```diff
@@ -202,6 +202,15 @@ defmodule Liquid.Combinators.General do
       string(@start_tag)
     ])
     |> reduce({List, :to_string, []})
+    |> traverse({Liquid.Combinators.General, :check_empty, []})
+  end
+
+  def check_empty(_rest, args, context, _line, _offset) do
```
Comment (Eptis) [inline lib/liquid/combinators/general.ex]: I think its more efficient to do the if-else als 2 different function headers, where you pattern match on `args` being `""`

File: lib/liquid/nimble_parser.ex [hunk 2]
```diff
@@ -144,14 +141,19 @@ defmodule Liquid.NimbleParser do
...
   """
   @spec parse(String.t()) :: {:ok | :error, any()}
+  def parse(""), do: {:ok, []}
+
   def parse(markup) do
     case __parse__(markup) do
       {:ok, template, "", _, _, _} ->
         {:ok, template}
 
+      {:error, mesagge, _, _, _, _} ->
```
Comment (Eptis) [inline lib/liquid/nimble_parser.ex]: `mesagge` -> `message`

File: lib/liquid/nimble_parser.ex [hunk 3]
```diff
@@ -144,14 +141,19 @@ defmodule Liquid.NimbleParser do
...
+        {:error, "#{mesagge}"}
+
       {:ok, _, rest, _, _, _} ->
         {:error, "Error parsing: #{rest}"}
```
Comment (Eptis) [inline lib/liquid/nimble_parser.ex]: What is `rest` in this case?

File: test/liquid/combinators/tags/assign_test.exs [hunk 4]
```diff
@@ -138,11 +126,11 @@ defmodule Liquid.Combinators.Tags.AssignTest do
...
   end
 
-  test "incorrect variable assignment" do
-    test_combinator_error("{% assign cart@ = 5 %}", &Parser.assign/1)
-    test_combinator_error("{% assign cart. = 5 %}", &Parser.assign/1)
-    test_combinator_error("{% assign .cart = 5 %}", &Parser.assign/1)
-    # TODO: this must fails
-    # test_combinator_error("{% assign cart? = 5 %}", &Parser.assign/1)
-  end
+  # test "incorrect variable assignment" do
```
Comment (Eptis) [inline test/liquid/combinators/tags/assign_test.exs]: Why is this commented out? either add `@skip true` to skip the test or remove it all together. This does look like a case we should test.

File: lib/liquid/nimble_parser.ex (hunk 3 above)
Comment (fabianherrera) [reply lib/liquid/nimble_parser.ex]: Rest contains all markup from error raises. It contains where error was found. (Could not be parsed)

File: lib/liquid/combinators/general.ex (hunk 1 above)
Comment (Spadavecchia) [reply lib/liquid/combinators/general.ex]: Done

File: lib/liquid/nimble_parser.ex (hunk 2 above)
Comment (Spadavecchia) [reply lib/liquid/nimble_parser.ex]: Done

File: lib/liquid/nimble_parser.ex (hunk 3 above)
Comment (Spadavecchia) [reply lib/liquid/nimble_parser.ex]: Renamed: `rest` -> `incorrect_markup`

File: test/liquid/combinators/tags/assign_test.exs (hunk 4 above)
Comment (Spadavecchia) [reply test/liquid/combinators/tags/assign_test.exs]: Done. Test was uncommented.

**Pull request merged**